Marlin/Marlin_main.cpp

```diff
@@ -2080,70 +2084,48 @@ static void clean_up_after_endstop_or_probe_move() {
     return current_position[Z_AXIS];
   }
 
-#endif // HAS_BED_PROBE
-
-#if HAS_PROBING_PROCEDURE
-
-  inline void do_blocking_move_to_xy(float x, float y) {
-    do_blocking_move_to(x, y, current_position[Z_AXIS]);
-  }
-
-  enum ProbeAction {
-    ProbeStay          = 0,
-    ProbeDeploy        = _BV(0),
-    ProbeStow          = _BV(1),
-    ProbeDeployAndStow = (ProbeDeploy | ProbeStow)
-  };
-
   // Probe bed height at position (x,y), returns the measured z value
-  static float probe_pt(float x, float y, float z_raise, ProbeAction probe_action = ProbeDeployAndStow, int verbose_level = 1) {
+  static float probe_pt(float x, float y, bool stow = true, int verbose_level = 1) {
     #if ENABLED(DEBUG_LEVELING_FEATURE)
       if (DEBUGGING(LEVELING)) {
         SERIAL_ECHOLNPGM("probe_pt >>>");
-        SERIAL_ECHOPAIR("> ProbeAction:", probe_action);
+        SERIAL_ECHOPAIR("> stow:", stow);
         SERIAL_EOL;
         DEBUG_POS("", current_position);
       }
     #endif
 
-    float old_feedrate = feedrate;
-
-    // Raise by z_raise, then move the Z probe to the given XY
-    #if ENABLED(DEBUG_LEVELING_FEATURE)
-      if (DEBUGGING(LEVELING)) {
-        SERIAL_ECHOPAIR("Z Raise by z_raise ", z_raise);
-        SERIAL_EOL;
-      }
-    #endif
-    do_probe_raise(z_raise); // this also updates current_position
-
     #if ENABLED(DEBUG_LEVELING_FEATURE)
       if (DEBUGGING(LEVELING)) {
-        SERIAL_ECHOPAIR("> do_blocking_move_to_xy ", x - (X_PROBE_OFFSET_FROM_EXTRUDER));
+        SERIAL_ECHOPAIR("> do_blocking_move_to ", x - (X_PROBE_OFFSET_FROM_EXTRUDER));
         SERIAL_ECHOPAIR(", ", y - (Y_PROBE_OFFSET_FROM_EXTRUDER));
+        SERIAL_ECHOPAIR(", ", max(current_position[Z_AXIS],Z_RAISE_BETWEEN_PROBINGS));
         SERIAL_EOL;
       }
     #endif
 
     // this also updates current_position
-    feedrate = XY_PROBE_FEEDRATE;
-    do_blocking_move_to_xy(x - (X_PROBE_OFFSET_FROM_EXTRUDER), y - (Y_PROBE_OFFSET_FROM_EXTRUDER));
+    do_blocking_move_to(x - (X_PROBE_OFFSET_FROM_EXTRUDER), y - (Y_PROBE_OFFSET_FROM_EXTRUDER), max(current_position[Z_AXIS],Z_RAISE_BETWEEN_PROBINGS));
```

Comment:
> Some machines, mostly deltas, have either a large raise-before or a large raise-after value.

As far as i know this is/was because of the bowl shape moves. They want to be far away from the bed when deploying/stowing. If you have a cloth look on the mechanics if the Allen key probes you will discover - the deploy move could be purely horizontal. The extender of the short side of the Allen key has to hit the belt or tower to turn the key from its rest. That does not need a vertical move. Stowing is different. At a special x/y place there the move is vertical downward until the spring can turn the key on its rest.
If we don't have the bowl shaped moves any more the very high raises are not needed. The raise shold be high enough to bring the pit of the Allen key above the stow platform. Oterwise, (we still use potentially diagonal moves) the tip could collide with the platform if the start point is cloth to the platform.

Ok. From the beginning.
For the deltas, an all other machines homing to z-max, we have not to deal with unhomed axis for deploying/stowing. We don't need a z-min-probe for homing to max.
As soon z is homed to max (we are still at the top) we know we are very high compared to our raise/min-height values - so we know we do not need to rise.   
A __raise__ move would be disastrous.
A unconditional move to a special high will probably be to deep. The current deploy height for the allen keys is at about the mid of the usable height of the towers.
A conditional move (like we do now) seems to be perfect. 

The Allen key moves need special attention. Right after homing we are still at the very top. Here we can not move in x/y-direction before we are a bit deeper - can not use do_blocking_move().
At the low end of the z axis the blocking moves are welcome.    

I see a problem with some of the default values for the Allen key moves. For example "mini kossel". Here we can find:
```
  // Kossel Mini
  #define Z_PROBE_ALLEN_KEY_DEPLOY_1_X 30.0
  #define Z_PROBE_ALLEN_KEY_DEPLOY_1_Y DELTA_PRINTABLE_RADIUS
  #define Z_PROBE_ALLEN_KEY_DEPLOY_1_Z 100.0
  #define Z_PROBE_ALLEN_KEY_DEPLOY_1_FEEDRATE XY_PROBE_SPEED
```
If `DELTA_PRINTABLE_RADIUS` is indeed the limit then the point can't be reached. a² + b² > a². The same schuld be true for the stow-platform-point. Ideally this should be outside the normal print area. Otherwise a collision while printing is likely. With working software endstops it will become a problem to reach this points.
The same i realized for the sled probe. As soon a positive `SLED_DOCKING_OFFSET` is defined the position can not be reached. For these moves we'd have to disable the software endstops.
Also 'cleaning area' should be protected but reachable.

Sorry for the monologue - you solved the problem by your self - an i agree.